backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java

```diff
@@ -50,65 +63,128 @@ private void updatePersonFacility(Person person, UUID facilityId) {
+  // don't trust an id provided by the user.
+  private UUID getOrgId() {
+    return _os.getCurrentOrganization().getInternalId();
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public long getPatientsCount(UUID facilityId) {
-    return _repo.countAllByFacilityAndOrganization(
-        _os.getFacilityInCurrentOrg(facilityId), _os.getCurrentOrganization(), false);
+  // Specifications filters for queries
+  private Specification<Person> inWholeOrganizationFilter() {
+    return (root, query, cb) ->
+        cb.equal(root.get(SpecField.Organization).get(SpecField.InternalId), getOrgId());
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public List<Person> getAllPatients(int pageOffset, int pageSize) {
-    return _repo
-        .findAllByOrganization(
-            _os.getCurrentOrganization(), false, PageRequest.of(pageOffset, pageSize, NAME_SORT))
-        .toList();
+  // Note: Patients with NULL facilityIds appear in ALL facilities.
+  private Specification<Person> inFacilityFilter(@NotNull UUID facilityId) {
+    return (root, query, cb) ->
+        cb.and(
+            cb.equal(root.get(SpecField.Organization).get(SpecField.InternalId), getOrgId()),
+            cb.or(
+                cb.isNull(root.get(SpecField.Facility)), // null check first
+                cb.equal(root.get(SpecField.Facility).get(SpecField.InternalId), facilityId)));
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public long getAllPatientsCount() {
-    return _repo.countAllByOrganization(_os.getCurrentOrganization(), false);
+  private Specification<Person> nameMatchesFilter(
+      @Size(min = MINIMUMCHARFORSEARCH) String namePrefixMatch) {
+    String likeString = namePrefixMatch.trim().toLowerCase() + "%";
+    return (root, query, cb) ->
+        cb.or(
+            cb.like(cb.lower(root.get(SpecField.PersonName).get(SpecField.FirstName)), likeString),
+            cb.like(cb.lower(root.get(SpecField.PersonName).get(SpecField.MiddleName)), likeString),
+            cb.like(cb.lower(root.get(SpecField.PersonName).get(SpecField.LastName)), likeString));
   }
 
-  // FYI archived is not just a parameter because it requires different permissions.
-  @AuthorizationConfiguration.RequirePermissionReadArchivedPatientList
-  public List<Person> getArchivedPatients(UUID facilityId, int pageOffset, int pageSize) {
-    return _repo
-        .findByFacilityAndOrganization(
-            _os.getFacilityInCurrentOrg(facilityId),
-            _os.getCurrentOrganization(),
-            true,
-            PageRequest.of(pageOffset, pageSize, NAME_SORT))
-        .toList();
+  private Specification<Person> isDeletedFilter(boolean isDeleted) {
+    return (root, query, cb) -> cb.equal(root.get(SpecField.IsDeleted), isDeleted);
+  }
+
+  // called by List function and Count function
+  private boolean checkPermissionsForListFunc(
+      UUID facilityId, boolean isArchived, String searchTerm) {
```

Comment:
here and everywhere else it is used, please use a variable more specific than `searchTerm`. it seems what you really mean is `namePrefixMatch`. it's fine to reuse that name!

Reply:
This is a pretty big rename since it's part of the graphql api, but it's important to get it as close to right as possible at this stage

Reply:
Counterpoint: we might in the future change the semantics of the searchTerm without wanting to change the API. I would leave it as searchTerm for the graphql query, but maybe change it at the service layer?